packages/app/src/request.ts

```diff
@@ -32,17 +37,64 @@ export const getProtocol = (req: Request, trust: Trust): Protocol => {
   return index !== -1 ? header.substring(0, index).trim() : header.trim()
 }
 
-export const getHostname = (req: Request, trust: Trust): string | undefined => {
-  let host: string = req.get('X-Forwarded-Host') as string
+const normalizeHostString = (host: string): string => decodeURIComponent(host).toLowerCase().normalize()
+
+const getAuthorityHeaderHostString = (req: Request): string | undefined => {
+  const authority = req.get(':authority')
+  if (Array.isArray(authority)) return undefined
+  if (!authority) return undefined
```

Comment:
can be simplified with 
```suggestion
  if (Array.isArray(authority) || !authority) return undefined
```